Client/Frontend/Settings/CustomSearchEngineForm.swift

```diff
@@ -0,0 +1,188 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import Shared
+import SnapKit
+import Storage
+import WebImage
+
+private let log = Logger.browserLogger
+
+class CustomSearchEngineForm: SettingsTableViewController {
+    
+    private var urlString: String?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = "Add Custom Search Engine"
+    }
+    
+    func faviconFor(url: NSURL) -> Favicon? {
+        var resultFavicon: Favicon?
```

Comment:
This works now, but now, when a favicon is found, it is downloaded by the `SDWebImageManager` before `self.profile.searchEngines.addSearchEngine` is passed the image and other details.
This causes a period of gap between when the user taps 'Save', and when the success alert is shown. Should an activity indicator be shown while the image downloads?